logger/logger.go

```diff
@@ -1,80 +1,173 @@
 package logger
 
-import "log"
+import (
+	"log"
+	"os"
+	"runtime"
+
+	"github.com/fatih/color"
+)
 
 // The Logger interface defines the levels a logging can occur at.
 type Logger interface {
-	Debug(msg string, data map[string]interface{})
-	Error(msg string, data map[string]interface{})
-	Fatal(msg string, data map[string]interface{})
-	Info(msg string, data map[string]interface{})
-	Warn(msg string, data map[string]interface{})
+	Debug(msg string, ctx *LogContext)
+	Error(msg string, ctx *LogContext)
+	Fatal(msg string, ctx *LogContext)
+	Info(msg string, ctx *LogContext)
+	Warn(msg string, ctx *LogContext)
+
+	LogLevel() LogLevel
 }
 
+type LogLevel int
+
 const (
-	colorRed    = "\x1b[31;1m"
-	colorBlue   = "\x1b[34;1m"
-	colorYellow = "\x1b[33;1m"
-	colorPink   = "\x1b[35;1m"
-	colorWhite  = "\x1b[37;1m"
-	colorClose  = "\x1b[0m"
-
-	logLevelDebug = "[DEBUG]"
-	logLevelInfo  = "[INFO]"
-	logLevelWarn  = "[WARN]"
-	logLevelError = "[ERROR]"
-	logLevelFatal = "[FATAL]"
+	LogLevelUnk LogLevel = iota
+	LogLevelDebug
+	LogLevelInfo
+	LogLevelWarn
+	LogLevelError
+	LogLevelFatal
 )
 
+func NewLogLevel(val string) LogLevel {
+	switch val {
+	case "DEBUG":
+		return LogLevelDebug
+	case "INFO":
+		return LogLevelInfo
+	case "WARN":
+		return LogLevelWarn
+	case "ERROR":
+		return LogLevelError
+	case "FATAL":
+		return LogLevelFatal
+	default:
+		return LogLevelUnk
+	}
+}
+
+func (ll LogLevel) String() string {
+	return map[LogLevel]string{
+		LogLevelDebug: "[DEBUG]",
+		LogLevelInfo:  "[INFO]",
+		LogLevelWarn:  "[WARN]",
+		LogLevelError: "[ERROR]",
+		LogLevelFatal: "[FATAL]",
+		LogLevelUnk:   "[UNK]",
+	}[ll]
+}
+
 // TrailsLogger implements Logger using log.
 type TrailsLogger struct {
-	l *log.Logger
+	skip int
+	env  string
+	l    *log.Logger
+	ll   LogLevel
 }
 
 // NewLogger constructs a TrailsLogger.
-func NewLogger() Logger { return &TrailsLogger{log.Default()} }
+//
+// Logs are printed to os.Stdout by default, using the std lib log pkg.
+// The default environment is DEVELOPMENT.
+// The default log level is DEBUG.
+func NewLogger(opts ...LoggerOptFn) Logger {
```

Comment:
Nitpick: but since you are going to be importing logger, you will call `logger.NewLogger`, which is redundant. I already have the "logger" info so I'm wanting to call `logger.New` here.